components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -1354,6 +1322,60 @@ private static String getTenantUrl(String url, String tenantDomain) throws URISy
         }
     }
 
+    /**
+     * Builds a URL with a given context in both the tenant-qualified url supported mode and the legacy mode. Returns
+     * the absolute URL build from the default context in the tenant-qualified url supported mode and returns the
+     * absolute url build from file configuration context.
+     *
+     * @param defaultContext              Default URL context.
+     * @param getValueFromFileBasedConfig File-based Configuration.
+     * @return Absolute URL.
+     */
+    public static String buildUrl(String defaultContext, Supplier<String> getValueFromFileBasedConfig) {
+
+        String oauth2EndpointURLInFile = null;
+        if (getValueFromFileBasedConfig != null) {
+            oauth2EndpointURLInFile = getValueFromFileBasedConfig.get();
+        }
+        try {
+            return buildServiceUrl(defaultContext, oauth2EndpointURLInFile);
+        } catch (URLBuilderException e) {
+            throw new IdentityRuntimeException("Error while building url for context: " + defaultContext);
```

Comment:
Ideally, if this is to be a public API, this should not throw a runtime exception. You should wrap this for an exception type defined in this module